front/production_checks/lib/libcheck.ts

```diff
@@ -0,0 +1,10 @@
+export type CheckFunction = (
+  checkName: string,
+  reportSuccess: (reportPayload: unknown) => void,
+  reportFailure: (reportPayload: unknown, message: string) => void
+) => Promise<void>;
+
+export type Check = {
+  name: string;
```

Comment:
intervalHours some we may not want to run hourly?

Reply:
Mentionned in the specs that we will do that in a follow up version of this framework. Just to get it quickly out of the door.